liquidctl/driver/krakenz3.py

```diff
@@ -0,0 +1,495 @@
+"""liquidctl drivers for fourth-generation NZXT Kraken X and Z liquid coolers.
+
+Supported devices:
+
+- NZXT Kraken X (X53, X63 and Z73)
+- NZXT Kraken Z (Z53, Z63 and Z73)
+
+Copyright (C) 2020–2022  Tom Frey, Jonas Malaco, Shady Nawara and contributors
+SPDX-License-Identifier: GPL-3.0-or-later
+"""
+# uses the psf/black style
+
+import os
+import io
+import math
+import pathlib
+import logging
+from time import sleep
+
+from PIL import Image, ImageSequence
+
+from liquidctl.driver.usb import UsbDriver, UsbHidDriver
+from liquidctl.util import (
+    normalize_profile,
+    interpolate_profile,
+    clamp,
+    Hue2Accessory,
+    HUE2_MAX_ACCESSORIES_IN_CHANNEL,
+    map_direction,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+_READ_LENGTH = 64
+_WRITE_LENGTH = 64
+_BULK_WRITE_LENGTH = 512
+_MAX_READ_ATTEMPTS = 12
+
+_STATUS_TEMPERATURE = "Liquid temperature"
+_STATUS_PUMP_SPEED = "Pump speed"
+_STATUS_PUMP_DUTY = "Pump duty"
+_STATUS_FAN_SPEED = "Fan speed"
+_STATUS_FAN_DUTY = "Fan duty"
+
+# Available speed channels for model Z coolers
+# name -> (channel_id, min_duty, max_duty)
+# TODO adjust min duty values to what the firmware enforces
+_SPEED_CHANNELS_KRAKENZ = {
+    "pump": (0x1, 20, 100),
+    "fan": (0x2, 0, 100),
+}
+
+_CRITICAL_TEMPERATURE = 59
+
+# Available color channels and IDs for model Z coolers
+_COLOR_CHANNELS_KRAKENZ = {"external": 0b001, "screen": 0b1111}
```

Comment:
I think it would cause issues with circular dependencies and would be confusing when half the constants are in one file and the other half here.

Reply:
It's true that a circular dependency will happen here because of the backward compatible alias that has to be added to `kraken3`.  However, it might be one of those circular dependencies that any modern CPython can handle, and I think it's at least worth testing.

Regarding splitting constants, I think you have a point when it comes to the types of constants that will indeed be split, say, for example, the status item names.

However, having really identical things like `_COLOR_MODES` duplicated in both modules is a lot of unnecessary noise, and it also induces readers to think that there is something different between the two (which can indeed happen, like in HUE2 vs. Kraken X|Z3, but isn't the case here).  And, naturally, there's also the potential for fixes to one not getting applied to the other.

So how about only moving the external HUE channel constants, as those are _all_ identical between modules, and that would already clear up most of noise?